test/timeparsing.cc

```diff
@@ -401,6 +401,29 @@ TEST(TimeParsing, TestConditionalRestrictions) {
       TryConditionalRestrictions(conditions.at(x), 0, 1, 64, 3, 3, 0, 15, 0, 12, 1, 5, 17, 0);
     }
   }
+
```

Comment:
Do we want to test wrongly formatted restriction...eg: end day is set without a corresponding start day (`-Friday` or some such)?

Reply:
As for me it sounds as incorrect input. In case only day_on is specified, '-' is not added, day_on can be a sequence of days separated by a comma or a range with '-' inside. When only day_off is specified it is harder to understand what it means: working hours are from Monday/Sunday to X, or each day except X....?

Reply:
You're right, in that the input is incorrect. My point was that we should test that such inputs are ignored (looking at the code, we only set the condition when day_start is set, so just having a day_end should not enable the TR). So adding a test with incorrect input and showing that it does not take effect would be sufficient. Does that make sense?

Reply:
Yeah, added regular and gurka tests.

Reply:
@CuriOusQuOkka @mandeepsandhu if we have only the start day set then these restrictions should be tossed as the user entered the info incorrectly